protocol/0075-PLUP_protocol_upgrades.md

```diff
@@ -0,0 +1,166 @@
+
+NB: by the time of the block height of the upgrade the validators must have downloaded and built the binaries and prepared them in the right location as required by the vega-visor.
+
+## Framework / data structures
+
+As protobuf:
+```go
+message ProtocolUpgradeProposal {
+   // The block height at which to perform the upgrade
+   uint64 upgrade_block_height = 1;
+   // the release tag for the vega binary
+   string vega_release_tag = 2;
+   // the release tag for the data-node binary
+   string data_node_release_tag = 3;
+}
+
+enum ProtocolUpgradeProposalStatus {
+  PROTOCOL_UPGRADE_PROPOSAL_STATUS_UNSPECIFIED = 0;
+  // The proposal is pending
+  PROTOCOL_UPGRADE_PROPOSAL_STATUS_PENDING = 1;
+  // The proposal is approved
+  PROTOCOL_UPGRADE_PROPOSAL_STATUS_APPROVED = 2;
+  // The proposal is rejected
+  PROTOCOL_UPGRADE_PROPOSAL_STATUS_REJECTED = 3;
+}
+
+message ProtocolUpgradeEvent {
+   // The block height at which to perform the upgrade
+   uint64 upgrade_block_height = 1;
+   // the release tag for the vega binary
+   string vega_release_tag = 2;
+   // the release tag for the data-node binary
+   string data_node_release_tag = 3;
+   // tendermint validators that have agreed to the upgrade
+  repeated string approvers = 4;
+  // the status of the proposal
+  ProtocolUpgradeProposalStatus status = 5;
+}
+
+```
+
+## Acceptance criteria
+
+### Invalid proposals - Rejections
+   - A network with 5 validators 
+   - (<a name="0075-PLUP-001" href="#0075-PLUP-001">0075-PLUP-001</a>) Validator proposes a protocol upgrade to an invalid tag (https://semver.org/) - should result in an error
+   - (<a name="0075-PLUP-002" href="#0075-PLUP-002">0075-PLUP-002</a>) Validator proposes a protocol upgrade on a block height preceding the current block - should result in an error
+   - (<a name="0075-PLUP-003" href="#0075-PLUP-003">0075-PLUP-003</a>) Propose and enact a version downgrade
+   - (<a name="0075-PLUP-004" href="#0075-PLUP-004">0075-PLUP-004</a>) Non-validator attempts to propose upgrade
+   - (<a name="0075-PLUP-005" href="#0075-PLUP-005">0075-PLUP-005</a>) Ersatz validator attempts to propose upgrade
+ 
+### Block height validation
+   Proposal will not be accepted as valid if validator
+   -  (<a name="0075-PLUP-006" href="#0075-PLUP-006">0075-PLUP-006</a>) Proposes a negative upgrade block
+   -  (<a name="0075-PLUP-007" href="#0075-PLUP-007">0075-PLUP-007</a>) Proposes a 0 upgrade block
+   -  (<a name="0075-PLUP-008" href="#0075-PLUP-008">0075-PLUP-008</a>) Proposes (string/other upgrade block)
+   -  (<a name="0075-PLUP-009" href="#0075-PLUP-009">0075-PLUP-009</a>) Proposes without supplying a block height
+
+### VISOR 
+   - (<a name="0075-PLUP-010" href="#0075-PLUP-010">0075-PLUP-010</a>) Can be seen to automatically download the latest version for install when available at the source location when file meets the format criteria defined
```

Comment:
The latest  version and the one in an accepted proposal may be different. Should we add an AC that the vegavisor downloads all the versions? Or all the ones in active proposal? Or is that not needed / intended behavior?